src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/InteropHelpers.cs

```diff
@@ -231,6 +231,31 @@ internal static unsafe string BstrBufferToString(char* buffer)
             return Marshal.PtrToStringBSTR((IntPtr)buffer);
         }
 
+        internal static unsafe byte* StringToAnsiBstrBuffer(string s)
+        {
+            if (s is null)
+            {
+                return (byte*)IntPtr.Zero;
+            }
+
+            int stringLength = s.Length;
+            fixed (char* pStr = s)
+            {
+                int nativeLength = PInvokeMarshal.GetByteCount(pStr, stringLength);
+                byte* bstr = (byte*)Marshal.AllocBSTR(nativeLength);
```

Comment:
I do not think that this works. `AllocBSTR` takes length in characters, but you are passing it length in bytes.